src/utils/stark/stark-curve.ts

```diff
@@ -28,3 +77,20 @@ export function fixMessage(msg: string) {
   // In this case delta will be 4 so we perform a shift-left of 4 bits by adding a ZERO_BN.
   return `${msg}0`;
 }
+
+function getXCoordinate(publicKey: string): string {
+  const keyPair = starkEc.keyFromPublic(encUtils.hexToArray(publicKey));
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return encUtils.sanitizeBytes((keyPair as any).pub.getX().toString(16), 2);
+}
+
+export function getStarkPublicKey(keyPair: ec.KeyPair): string {
+  const pub = keyPair.getPublic(true, 'hex');
+  return encUtils.sanitizeHex(getXCoordinate(pub));
+}
+
+// Generates a new private key on the Stark-friendly elliptic curve.
+export function generatePrivateKey(): string {
```

Comment:
Inspired by: https://github.com/ethereum/go-ethereum/blob/master/crypto/crypto.go#L256

Reply:
If we decide for not allowing developers to input their own seed it means we need to duplicate this whole Stark Curve configuration in the Core and Wallet SDKs. I'd rather have the seed option as optional in the core, so Wallet SDK and other developers can pass in whatever seed they feel like, otherwise if seed is null we can default to random.

Reply:
Perhaps we should export the curve itself, so that the Wallet SDK can use it easily?

It's not just having the seed option @CassiusPacheco - in the Link we use an extremely custom mechanism for going from seed --> key involving BIP 32 paths and a grinding algorithm, it's not just "here's a random seed, generate a key". In my mind that mechanism belongs in the wallet SDK. Happy to explain my thoughts further?

Reply:
Yeah, I'm aware of the derivation, I just thought we could do the derivation in the Wallet and just pass in the ground key to the core as a seed to generate the keypair, but talking to @Kastet I realised we're using a completely different method here called `genKeyPair`, instead of the former `keyFromPrivate` that the wallet relies on, so my previous commentary doesn't make sense.

In that case, no need to export the curve, it'll be little value, I believe we should just handle everything related to that in the Wallets SDK directly, including `keyFromPrivate`. No problem!

Reply:
@alex-connolly we've just discussed it with @CassiusPacheco. There are two options:
1. (what this PR is proposing) Core SDK only exposes: 
 - `generatePrivateKey` for use cases without Wallet SDK.
 - `StandardStarkSigner` class that's initialised with a `ec.KeyPair` created by the Wallet.

This requires Wallet SDK to also have a copy of the stark curve.

2. In addition to the above, Core SDK exposes the curve so that the Wallet SDK doesn't need to have a copy of it to create a keypair 
```
  const keyPair = starkEc.keyFromPrivate(privateKey, 'hex');
```
where `privateKey` is the result of our custom derivations and grinding.


Removes the need for the Wallet SDK to have a copy of the curve but required the curve to be exposed by the Core SDK.

I'm fine with any of these because we're talking about moving extra 30 lines of code to set up the curve that never change.

If the [Stark package](https://github.com/starkware-libs/starkware-crypto-utils) had TS support then both SDK could just import it (would increase bundle size though)

Reply:
Going with the 1st option. The curve can be easily added later to the interface.